addon/search/search.js

```diff
@@ -76,19 +116,18 @@
     var state = getSearchState(cm);
     if (state.query) return findNext(cm, rev);
     var q = cm.getSelection() || state.lastQuery;
-    dialog(cm, queryDialog, "Search for:", q, function(query) {
+    dialog(cm, queryDialog, "Search for:", q, function(queryStr, keyEvent) {
       cm.operation(function() {
-        if (!query || state.query) return;
-        state.query = parseQuery(query);
-        cm.removeOverlay(state.overlay, queryCaseInsensitive(state.query));
-        state.overlay = searchOverlay(state.query, queryCaseInsensitive(state.query));
-        cm.addOverlay(state.overlay);
-        if (cm.showMatchesOnScrollbar) {
-          if (state.annotate) { state.annotate.clear(); state.annotate = null; }
-          state.annotate = cm.showMatchesOnScrollbar(state.query, queryCaseInsensitive(state.query));
+        initQuery(cm, queryStr);
+
+        if (cm.state.incrementalSearch) {
+          // In incremental search we need to prevent default behavior
+          // from pre-maturely closing the dialog box.
+          CodeMirror.e_stop(keyEvent);
+          findNext(cm, (keyEvent.shiftKey === true));
```

Comment:
Rather than providing custom keybindings for our search dialog, rely on the fact the dialog binds Enter to the execution of this callback.  I don't like this.

Another option is to override the dialog's keydown event and specify search specific bindings.

Other thoughts or opinions?